src/portable/st/stm32f4/dcd_stm32f4.c

```diff
@@ -80,20 +80,39 @@ static void bus_reset(void) {
   dev->DOEPMSK |= USB_OTG_DOEPMSK_STUPM | USB_OTG_DOEPMSK_XFRCM;
   dev->DIEPMSK |= USB_OTG_DIEPMSK_TOM | USB_OTG_DIEPMSK_XFRCM;
 
+  // Peripheral FIFO architecture (Rev18 RM 29.11)
+  //
+  // --------------- 312.5 ( 1250 bytes )
+  // | IN FIFO 3  |
+  // --------------- y + x + 16 + GRXFSIZ
+  // | IN FIFO 2  |
+  // --------------- x + 16 + GRXFSIZ
+  // | IN FIFO 1  |
+  // --------------- 16 + GRXFSIZ
+  // | IN FIFO 0  |
+  // --------------- GRXFSIZ
+  // | OUT FIFO   |
+  // | ( Shared ) |
+  // --------------- 0
+  //
   // FIFO sizes are set up by the following rules (each word 32-bits):
-  // OUT FIFO uses (based on page 1354 of Rev 17 of reference manual):
+  // All EP OUT shared a unique OUT FIFO which uses (based on page 1354 of Rev 17 of reference manual):
   // * 10 locations in hardware for setup packets + setup control words
   // (up to 3 setup packets).
   // * 2 locations for OUT endpoint control words.
-  // * 64 bytes for maximum control packet size.
+  // * 16 + 1 (data + info) for largest packet size of 64 bytes.
```

Comment:
No need for the `+1` for `info`. I already account for the `info`  when I said "2 locations for OUT endpoint control words."

Reply:
ah thanks, it isn't so clear to me from the RM :D